tests/Aggregation/DateHistogramTest.php

```diff
@@ -18,7 +18,62 @@ class DateHistogramTest extends BaseAggregationTest
      */
     public function testDateHistogramAggregation(): void
     {
-        $agg = new DateHistogram('hist', 'created', '1h');
+        $this->_checkVersion('7.2');
+
+        $agg = new DateHistogram('hist', 'created');
+        $agg->setFixedInterval('1h');
+
+        $query = new Query();
+        $query->addAggregation($agg);
+        $results = $this->_getIndexForTest()->search($query)->getAggregation('hist');
+
+        $docCount = 0;
+        $nonDocCount = 0;
+        foreach ($results['buckets'] as $bucket) {
+            if (1 == $bucket['doc_count']) {
+                ++$docCount;
+            } else {
+                ++$nonDocCount;
+            }
+        }
+        // 3 Documents that were added
+        $this->assertEquals(3, $docCount);
+        // 1 document that was generated in between for the missing hour
+        $this->assertEquals(1, $nonDocCount);
+    }
+
+    /**
+     * @group unit
+     * @group legacy
+     */
+    public function testDateHistogramAggregationWithIntervalTriggersADeprecation(): void
+    {
+        $this->expectDeprecation('Since ruflin/elastica 7.1.0: Argument 3 passed to "__construct()" is deprecated, use "setDateInterval()" or "setCalendarInterval()" instead. It will be removed in 8.0.');
+        new DateHistogram('hist', 'created', 'day');
+    }
+
+    /**
+     * @group unit
+     * @group legacy
+     */
+    public function testDateHistogramAggregationSetIntervalTriggersADeprecation(): void
+    {
+        $agg = new DateHistogram('hist', 'created');
+
+        $this->expectDeprecation('Since ruflin/elastica 7.1.0: The "setInterval()" method is deprecated, use "setDateInterval()" or "setCalendarInterval()" instead. It will be removed in 8.0.');
```

Comment:
This does not work @deguif 
Do you have an idea why?

Reply:
You don't call the `setInterval()`method, or am I missing something?
By the way __METHOD__ generate a full namespaced method name.

Reply:
Ok tested locally, it misses the inclusion of Symfony provided trait for deprecations.
See https://github.com/ruflin/Elastica/blob/master/tests/IndexTest.php#L27